parsers/US_NEISO.py

```diff
@@ -138,8 +147,98 @@ def fetch_production(zone_key='US-NEISO', session=None, target_datetime=None, lo
+    """Requests the last known power exchange (in MW) between two zones
+
+    Arguments:
+    zone_key1, zone_key2: specifies which exchange to get
+    session (optional): request session passed in order to re-use an existing session
+    target_datetime: the datetime for which we want production data. If not provided, we should
+      default it to now. The provided target_datetime is timezone-aware in UTC.
+    logger: an instance of a `logging.Logger`; all raised exceptions are also logged automatically
+
+    Return:
+    A list of dictionaries in the form:
+    [{
+      'sortedZoneKeys': 'CA-QC->US-NEISO',
+      'datetime': '2017-01-01T00:00:00Z',
+      'netFlow': 0.0,
+      'source': 'mysource.com'
+    }]
+    """
+    sorted_zone_keys = '->'.join(sorted([zone_key1, zone_key2]))
+
+    # For directions, note that ISO-NE always reports its import as negative
+
+    if sorted_zone_keys == 'CA-NB->US-NEISO':
+        # CA-NB->US-NEISO means import to NEISO should be positive
+        multiplier = -1
+
+        postdata = {
+            '_nstmp_zone0': '4010'  # ".I.SALBRYNB345 1"
+        }
+
+    elif sorted_zone_keys == "CA-QC->US-NEISO":
+        # CA-QC->US-NEISO means import to NEISO should be positive
+        multiplier = -1
+
+        postdata = {
+            '_nstmp_zone0': '4012',  # ".I.HQ_P1_P2345 5"
+            '_nstmp_zone1': '4013'  # ".I.HQHIGATE120 2"
+        }
+
+    elif sorted_zone_keys == 'US-NEISO->US-NY':
+        # US-NEISO->US-NY means import to NEISO should be negative
+        multiplier = 1
+
+        postdata = {
+            '_nstmp_zone0': '4014',  # ".I.SHOREHAM138 99"
+            '_nstmp_zone1': '4017',  # ".I.NRTHPORT138 5"
+            '_nstmp_zone2': '4011'  # ".I.ROSETON 345 1"
+        }
+
+    else:
+        raise Exception('Exchange pair not supported: {}'.format(sorted_zone_keys))
+
+    postdata['_nstmp_requestType'] = 'externalflow'
+
+    exchange_data = get_json_data(target_datetime, postdata, session)
+
+    summed_exchanges = defaultdict(int)
+    for exchange_key, exchange_values in exchange_data.items():
+        # sum up values from separate "exchanges" for the same date.
+        # this works because the timestamp of exchanges is always reported
+        # in exact 5-minute intervals by the API,
+        # e.g. "2018-03-18T00:05:00.000-04:00"
+        for datapoint in exchange_values:
+            dt = timestring_converter(datapoint['BeginDate'])
+            summed_exchanges[dt] += datapoint['Actual']
+
+    result = [
+        {
+            'datetime': timestamp,
+            'sortedZoneKeys': sorted_zone_keys,
+            'netFlow': value * multiplier,
+            'source': 'iso-ne.com'
+        }
+        for timestamp, value in summed_exchanges.items()
+    ]
+
+    return result
+
+
 if __name__ == '__main__':
     """Main method, never used by the Electricity Map backend, but handy for testing."""
 
+    from pprint import pprint
```

Comment:
Like the use of pprint for readability, I suppose importing in line avoids the cost when the module isn't run as main?

Reply:
I like importing it here to indicate that it's only used in the testing method. I doubt it makes any performance difference. Pylint doesn't complain about it, so I guess it's okay?

Reply:
I think it's fine.